### ISSUE TITLE
In class LGMLoss(nn.Module), which is the implementation of the original paper the covariance was considered as identity matrix instead of a diagonal matrix

### ISSUE THREAD

**Issue by kalyanainala** [YirongMao/softmax_variants#6]
In the following two lines from class LGMLoss(nn.Module),
log_covs = torch.unsqueeze(self.log_covs, dim=0)
covs = torch.exp(log_covs) # 1*c*d

the covariance matrix is considered as an identity matrix even though this is the implementation of the original paper whereas in the class LGMLoss_v0(nn.Module), (mentioned as  'LGMLoss whose covariance is fixed as Identity matrix' in the comments) no covariance matrix was considered.
